src/tests/functional/shared_test_classes/src/single_op/multiclass_nms.cpp

```diff
@@ -0,0 +1,184 @@
+        } else if (i == 0) { // bboxes
+            tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
+        } else { // roisnum
+            /* sum of rois is no larger than num_bboxes. */
+            ASSERT_TRUE(targetInputStaticShapes[i].size() == 1) << "Expected shape size 1 for input roisnum, got: " << targetInputStaticShapes[i];
+
+            // returns num random values whose sum is max_num.
+            auto _generate_roisnum = [](int num, int max_num) {
+                std::vector<int> array;
+                std::vector<int> results(num);
+
+                array.push_back(0);
+                for (auto i = 0; i < num-1; i++) {
+                    array.push_back(std::rand() % max_num);
+                }
+                array.push_back(max_num);
+
+                std::sort(array.begin(), array.end());
+
+                for (auto i = 0; i < num; i++) {
+                    results[i] = array[i+1] - array[i];
+                }
+
+                return results;
+            };
+            auto roisnum = _generate_roisnum(targetInputStaticShapes[i][0], targetInputStaticShapes[0][1]/*num_bboxes*/);
+
+            tensor = ov::Tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
+            if (tensor.get_element_type() == ov::element::i32) {
+                auto dataPtr = tensor.data<int32_t>();
+                for (size_t i = 0; i < roisnum.size(); i++) {
+                    dataPtr[i] = static_cast<int32_t>(roisnum[i]);
+                }
+            } else {
+                auto dataPtr = tensor.data<int64_t>();
+                for (size_t i = 0; i < roisnum.size(); i++) {
+                    dataPtr[i] = static_cast<int64_t>(roisnum[i]);
+                }
+            }
+        }
+
+        inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+    }
+}
+
+void MulticlassNmsLayerTest::SetUp() {
+    std::vector<InputShape> shapes;
+    InputTypes input_types;
+    size_t maxOutBoxesPerClass, backgroundClass, keepTopK;
+    element::Type outType;
+
+    op::util::MulticlassNmsBase::SortResultType sortResultType;
+
+    InputfloatVar inFloatVar;
+    InputboolVar inboolVar;
+    ov::op::util::MulticlassNmsBase::Attributes attrs;
+
+    std::tie(shapes, input_types, maxOutBoxesPerClass, inFloatVar, backgroundClass, keepTopK, outType, sortResultType, inboolVar, targetDevice)
+        = this->GetParam();
+
+    init_input_shapes(shapes);
+
+    ElementType paramsPrec, roisnumPrec, maxBoxPrec, thrPrec;
+    std::tie(paramsPrec, roisnumPrec, maxBoxPrec, thrPrec) = input_types;
+
+    float iouThr, scoreThr, nmsEta;
+    std::tie(iouThr, scoreThr, nmsEta) = inFloatVar;
+
+    bool sortResCB, normalized;
+    std::tie(sortResCB, normalized) = inboolVar;
+
+    ParameterVector params;
+    if (inputDynamicShapes.size() > 2) {
```

Comment:
there are only 2 cases: size == 2 and == 3. So `params` may be filled like `params = {shared_ptr(paramsPrec, ...), shared_ptr(paramsPrec, ...)}` and if size == 3, `params.push_back(shared_ptr(roisnumPrec, ...))`